packages/react/src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import MuiAutocomplete, {AutocompleteProps as MuiAutocompleteProps} from '@mui/material/Autocomplete';
+import clsx from 'clsx';
+import {forwardRef, ForwardRefExoticComponent, ReactElement, MutableRefObject} from 'react';
+import {WithWrapperProps} from '../../models';
+import {composeComponentDisplayName} from '../../utils';
+
+export type AutocompleteProps<T> = MuiAutocompleteProps<
```

Comment:
Do we need the generics here.
IMO we can simply do.
```suggestion
export type AutocompleteProps = MuiAutocompleteProps;
```

Reply:
Yes, we have to be specific, because `MUIAutocompleteProps` is giving an error when assigned directly. Using this we do not need to specifically pass the T when creating the component as it is a child of the `object`.

Reply:
Object is an antipattern now typescript. We need to use the `Record` type instead.

Reply:
Try this instead.

```ts
import MuiAutocomplete, {AutocompleteProps as MuiAutocompleteProps} from '@mui/material/Autocomplete';
import clsx from 'clsx';
import {forwardRef, ForwardRefExoticComponent, ReactElement, MutableRefObject, ElementType} from 'react';
import {WithWrapperProps} from '../../models';
import {composeComponentDisplayName} from '../../utils';

type AutocompleteProps<T> = MuiAutocompleteProps<T, boolean, boolean, boolean, ElementType>;

const COMPONENT_NAME: string = 'Autocomplete';

const Autocomplete: ForwardRefExoticComponent<AutocompleteProps<any>> & WithWrapperProps = forwardRef(
  (props: AutocompleteProps<any>, ref: MutableRefObject<HTMLDivElement>): ReactElement => {
    const {className, ...rest} = props;

    const classes: string = clsx('oxygen-autocomplete', className);

    return <MuiAutocomplete className={classes} {...rest} ref={ref} />;
  },
) as ForwardRefExoticComponent<AutocompleteProps<any>> & WithWrapperProps;

Autocomplete.displayName = composeComponentDisplayName(COMPONENT_NAME);
Autocomplete.muiName = COMPONENT_NAME;
Autocomplete.defaultProps = {};

export default Autocomplete;
```

Reply:
Fixed with https://github.com/wso2/oxygen-ui/pull/165/commits/4383e78c0c82fb2b57851a0822c130e8aa2dfc32